CRM/Report/Form/Member/Detail.php

```diff
@@ -254,6 +259,17 @@ public function __construct() {
     parent::__construct();
   }
 
+  public function where() {
+    parent::where();
+    $dbAlias = $this->_columns['civicrm_membership']['alias'];
+    if ($this->_params['primary_member_value'] == 1) {
+      $this->_where .= ' AND ( ' . $dbAlias . '.owner_membership_id IS NULL )';
+    }
+    if ($this->_params['primary_member_value'] != '' && $this->_params['primary_member_value'] == 0) {
```

Comment:
perhaps just 
```
$this->_params['primary_member_value'] === 0
```

ie use the triple equals for exact comparison & you don't need to check ''